sale_triple_discount/models/sale_order_line.py

```diff
@@ -11,22 +11,53 @@
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
-    @api.depends('discount2', 'discount3')
+    def _get_final_discount(self):
+        self.ensure_one()
+        if self.discounting_type == "additive":
+            return self._additive_discount()
+        elif self.discounting_type == "multiplicative":
+            return self._multiplicative_discount()
+
+    def _additive_discount(self):
+        self.ensure_one()
+        discounts = [getattr(self, x) or 0.0 for x in self._discount_fields()]
+        final_discount = 1 - sum(discounts) / 100
+        if final_discount <= 0:
+            return 0
+        return final_discount
+
+    def _multiplicative_discount(self):
+        self.ensure_one()
+        discounts = [1 - (getattr(self, x) or 0.0) / 100
```

Comment:
It's safer to use `self[x]` to invoke the fields since `getattr` could give access to the object methods.